android/app/src/main/java/org/fossasia/openevent/viewmodels/SearchActivityViewModel.java

```diff
@@ -0,0 +1,71 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.realm.RealmResults;
+
+public class SearchActivityViewModel extends ViewModel {
+
+    private MutableLiveData<List<Object>> searchResults;
+    private RealmDataRepository realmRepo;
+    private String searchText = "";
+
+    public SearchActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<List<Object>> getSearchResults() {
+        RealmResults<Microlocation> realmResults_microloc;
+        RealmResults<Track> realmResults_track;
+        RealmResults<Session> realmResults_session;
+        RealmResults<Speaker> realmResults_speaker;
+        List<Object> objectList = new ArrayList<>();
+
+        if (searchResults == null) {
+            searchResults = new MutableLiveData<>();
+            realmResults_track = realmRepo.getTracks();
+            objectList.add(realmResults_track);
+
+            int a = realmResults_track.size();
+            realmResults_session =realmRepo.getSessionsFiltered(0, searchText);
+            objectList.add(realmResults_session);
+
+            int b = realmResults_session.size();
+            int c = 0;
+            if(searchText!="") {
+                realmResults_speaker = realmRepo.getSpeakers(searchText);
+                objectList.add(realmResults_speaker);
+                c = realmResults_speaker.size();
+            }
+            realmResults_microloc = realmRepo.getLocations();
+            objectList.add(realmResults_microloc);
+
+            int d = realmResults_track.size();
+            if(a+b+c+d == 0){
+                return new MutableLiveData<>();
+            }else{
+                 searchResults.setValue(objectList);
+            }
+        }
```

Comment:
The naming is very poor here. Please change it to meaningful names rather than using a b c d e

There's no clue what the variables are used for.

Also, please see that the if cases are correctly used. You are doing everything in null check branch, whereas if the searchResults is not null, none of this will be called and what you will set as a value of objectList will be empty. Null check branch should only be used to instantiate the searchResults object and nothing else

Reply:
Also, in case of the sum being 0, you are sending a new MutableLiveData which is not needed. You should just set the empty value to the existing LiveData

Reply:
Also please add spaces where necessary, like around the `else`